fiftyone/core/collections.py

```
@@ -10548,42 +10523,115 @@ def create_index(self, field_or_spec, unique=False, wait=True, **kwargs):
+        if convert_to_unique:
+            logger.info(
+                f"Converting existing index '{index_name}' to unique "
+                f"on dataset '{self._dataset.name}'"
             )
 
-        name = coll.create_index(index_spec, unique=unique, **kwargs)
+            try:
+                index = {"name": _existing_db_name, "unique": True}
+                foo.get_db_conn().command("collMod", coll_name, index=index)
+
+                return _existing_name
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)